### PR TITLE
Fixes blanking of non-query atom data when QueryAtomData was being pi…

### PR DESCRIPTION
**Pull request by bp-kelley**
…ckled

#### What does this implement/fix? Explain your changes.
Fixes atom properties not being pickled properly when QueryAtomData was being set.

QueryAtomData is now deprecated since it does the same as AtomProps (they are both set to the same value now).  The QueryAtomData reader was retained (we might just drop it) but the writer was removed.

### CODE REVIEW

**Comment by bp-kelley** [comment]
@greglandrum This is a bug fix that should probably be included in the next release.

**Comment by greglandrum** [comment]
Yep. I'll take a look tomorrow morning.

File: Code/GraphMol/MolPickler.h [hunk 1]
```diff
@@ -45,12 +45,12 @@ class MolPicklerException : public std::exception {
...
 typedef enum {
   NoProps = 0,     // no data pickled
-  MolProps = 0x1,  // only public non computed properties
-  AtomProps = 0x10,
-  BondProps = 0x100,
-  QueryAtomData = 0x100,
-  PrivateProps = 0x10000,
-  ComputedProps = 0x100000,
+  MolProps      = BOOST_BINARY(1),  // only public non computed properties
+  AtomProps     = BOOST_BINARY(10),
```
Comment (greglandrum) [inline Code/GraphMol/MolPickler.h]: Doesn't this block of changes end up breaking backwards compatibility (i.e. old pickles will no longer be readable)?

Comment (bp-kelley) [reply Code/GraphMol/MolPickler.h]: I don't think so, these values are not written to the stream, the Tags::enum are.

Never the less, I'll make a regression test!

Comment (greglandrum) [reply Code/GraphMol/MolPickler.h]: Ah, ok, now I see how this is done.
Having a regression test that loads a molecule serialized with the current version is probably still a good idea though (I guess that's what you're thinking of).

**Comment by bp-kelley** [comment]
@greglandrum just added the unpickling old version test

**Comment by greglandrum** [comment]
I will merge as soon as the tests pass.

**Comment by greglandrum** [comment]
That py3 travis failure is a real (but easy) one.

**Comment by bp-kelley** [comment]
Just noticed that.

**Pull request merged**